crates/ef-testing/src/sequencer/setup.rs

```diff
@@ -0,0 +1,263 @@
+        );
+        storage.push(owner_storage);
+
+        // Initialize the bytecode storage var.
+        let bytecode_storage = &mut split_bytecode_to_starkfelt(bytecode)
+            .into_iter()
+            .enumerate()
+            .map(|(i, bytes)| {
+                (
+                    get_storage_var_address("bytecode_", &[StarkFelt::from(i as u32)]).unwrap(), // safe unwrap: var is ASCII
+                    bytes,
+                )
+            })
+            .collect();
+        let bytecode_len_storage = (
+            get_storage_var_address("bytecode_len_", &[]).unwrap(), // safe unwrap: var is ASCII
+            StarkFelt::from(bytecode.len() as u32),
+        );
+        storage.append(bytecode_storage);
+        storage.push(bytecode_len_storage);
+
+        // Initialize the kakarot address.
+        let kakarot_address_storage = (
+            get_storage_var_address("kakarot_address", &[]).unwrap(), // safe unwrap: var is ASCII
+            contract_address_to_starkfelt(&KAKAROT_ADDRESS),
+        );
+        storage.push(kakarot_address_storage);
+
+        // Initialize the nonce storage var.
+        let nonce = StarkFelt::from(TryInto::<u128>::try_into(nonce).map_err(|err| {
+            StarknetApiError::OutOfRange {
+                string: err.to_string(),
+            }
+        })?);
+        let nonce_storage = (
+            get_storage_var_address("nonce", &[]).unwrap(), // safe unwrap: var is ASCII
+            nonce,
+        );
+        storage.push(nonce_storage);
+
+        // Initialize the storage vars.
+        let is_evm_storage_empty = evm_storage.is_empty();
+        let evm_storage_storage = &mut evm_storage
+            .into_iter()
+            .flat_map(|(k, v)| {
+                let keys =
+                    get_uint256_storage_var_addresses("storage_", &split_u256(k).map(Into::into))
+                        .unwrap();
+                let values = split_u256(v).map(Into::into);
+                vec![(keys.0, values[0]), (keys.1, values[1])]
+            })
+            .collect();
+        storage.append(evm_storage_storage);
+
+        // Initialize the implementation.
+        let implementation_storage = (
+            get_storage_var_address("_implementation", &[]).unwrap(), // safe unwrap: var is ASCII
+            if bytecode.is_empty() && is_evm_storage_empty {
+                class_hash_to_starkfelt(&EOA_CLASS_HASH)
+            } else {
+                class_hash_to_starkfelt(&CONTRACT_ACCOUNT_CLASS_HASH)
+            },
+        );
+        storage.push(implementation_storage);
+
+        // Write all the storage vars to the sequencer state.
+        let starknet_address = starknet_address.try_into()?;
+        for (k, v) in storage {
+            (&mut self.0.state).set_storage_at(starknet_address, k, v);
+        }
+
+        // Set up the contract class hash and nonce.
+        self.0.state.set_nonce(starknet_address, Nonce(nonce));
```

Comment:
so you indeed set up nonce both in storage and in starknet state to avoid making an if/else?

Reply:
Yes, figured it wasn't a huge overhead. Actually the whole `initialize_contract` function is account agnostic and works for both accounts type, only if/else needed was for the `_implementation` storage var. Figured this was easier over having different setups per account type.

Reply:
you do one if/else, and I'm not really in favor of creating a setup that is not 100% identical to what it's supposed to be

Reply:
so you prefer one big if else based on the account then?